pandas/core/internals/managers.py

```diff
@@ -345,6 +345,37 @@ def _verify_integrity(self):
                 "tot_items: {1}".format(len(self.items), tot_items)
             )
 
+    def reduce(self, func, *args, **kwargs):
+        # If 2D, we assume that we're operating column-wise
+        if self.ndim == 1:
+            # we'll be returning a scalar
+            blk = self.blocks[0]
+            return func(blk.values, *args, **kwargs)
+
+        res = {}
+        for blk in self.blocks:
+            bres = func(blk.values, *args, **kwargs)
+            if np.ndim(bres) == 0 and blk.shape[0] != 1:
+                # i.e. we reduced over all axes and not just one; re-do column-wise
```

Comment:
I don't quite understand this case. How do we get here? re-calling `func` doesn't seem ideal.

Reply:
IIRC it was when we have axis=None

Reply:
after some digging, this appears to be coming from nanops funcs that are either not getting axis passed or are not handling it correctly.

Reply:
updated to make this check unnecessary